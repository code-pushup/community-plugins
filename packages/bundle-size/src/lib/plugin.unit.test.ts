import {describe, expect, it} from "vitest";
import {pluginConfigSchema} from "@code-pushup/models";
import bundleSizePlugin from "./plugin";

describe('bundleSizePlugin', () => {

    it('should create valid bundle-size plugin configuration', async () => {
      const bundleSizePluginConfig = bundleSizePlugin({});
        expect(() => pluginConfigSchema.parse(bundleSizePluginConfig)).not.toThrow();
        expect(bundleSizePluginConfig).toMatchSnapshot();
    });

});
