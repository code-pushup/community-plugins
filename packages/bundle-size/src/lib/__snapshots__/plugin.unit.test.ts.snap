// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`bundleSizePlugin > should create valid bundle-size plugin configuration 1`] = `
{
  "audits": [
    {
      "description": "Bundle size audit",
      "slug": "bundle-size",
      "title": "Bundle size",
    },
  ],
  "description": "Bundle size plugin for Code PushUp",
  "icon": "folder-css",
  "runner": [Function],
  "slug": "bundle-size",
  "title": "Bundle size",
}
`;
